File: backoffice/src/app/features/moderation/models/reportDto.ts
/** Signalement affiché dans la file de modération (GET /api/admin/reports). */
export interface ReportDto {
  id: string;
  /** 'post' | 'comment' (cible du signalement). */
  targetType: string;
  targetId: string;
  reason: string;
  status: string;
  reporterName: string;
  createdAt: string;
  /** Aperçu du contenu signalé (caption de post ou texte de commentaire). */
  targetPreview?: string | null;
  /** Image du post signalé (le cas échéant). */
  targetImageUrl?: string | null;
  /** Auteur du contenu signalé. */
  targetAuthorName?: string | null;
}

File: api/src/EventHub.Domain/ReadModels/ReportDto.cs
namespace EventHub.Domain.ReadModels;

/// <summary>Élément de la file de signalements affichée au back office.</summary>
public sealed record ReportDto(
    Guid Id,
    string TargetType,
    Guid TargetId,
    string Reason,
    string Status,
    string ReporterName,
    DateTime CreatedAt,
    // Aperçu du contenu signalé (pour que le modérateur le voie avant d'agir).
    string? TargetPreview = null,
    string? TargetImageUrl = null,
    string? TargetAuthorName = null);
